pennylane/estimator/templates/subroutines.py

```diff
@@ -971,6 +971,204 @@ def resource_decomp(cls, base_cmpr_op, num_iter):
         return gate_counts
 
 
+class UnaryIterationQPE(ResourceOperator):
+    r"""Resource class for UnaryIterationQPE. This is a variant of the Quantum Phase Estimation algorithm
+    that uses the unary iteration technique described in `arXiv.2011.03494 <https://arxiv.org/pdf/2011.03494>`_.
+    In this approach, the ``Select`` operation is implemented via unary iteration, which reduces
+    T and Toffoli gate counts at the cost of increasing the number of auxiliary qubits required. For
+    a detailed explanation of its typical usage, see the documentation `here <https://pennylane.ai/compilation/unary-iteration>`_.
+
+    Note, users can also provide a custom adjoint QFT implementation, which can be used to
+    further optimize the resource requirements. For example, one could use an approximate QFT implementation.
+
+    Args:
+        walk_operator (:class:`~.pennylane.estimator.resource_operator.ResourceOperator`): the walk operator to run
+            the phase estimation protocol on
+        num_iterations (int): The total number of times the walk operator :math:`W` is
+            applied in order to reach a target precision in the eigenvalue estimate.
+        adj_qft_op (:class:`~.pennylane.estimator.resource_operator.ResourceOperator` | None): An optional
+            argument to set the subroutine used to perform the adjoint QFT operation.
+        wires (WiresLike | None): the wires the operation acts on
+
+    Resources:
+        The resources are obtained from Fig 2. in Section III of `arXiv.2011.03494 <https://arxiv.org/pdf/2011.03494>`_.
+
+    .. seealso:: Related PennyLane operation :class:`~.pennylane.QPE` and explanation of `Unary Iteration <https://pennylane.ai/compilation/unary-iteration>`_.
+
+    **Example**
+
+    The resources for this operation are computed as follows:
+
+    >>> import pennylane.estimator as qre
+    >>> walk_operator = qre.RX()
```

Comment:
Maybe this can be an example using QubitizeTHC
```suggestion
    >>> thc_ham = qre.THCHamiltonian(num_orbitals=20, tensor_rank=40)
    >>> prep = qre.PrepTHC(thc_ham, coeff_precision=20, select_swap_depth=2)
    >>> walk_operator = qre.QubitizeTHC(thc_ham, prep_op=prep)
```